src/mono/mono/metadata/marshal-lightweight.c

```diff
@@ -2322,7 +2322,6 @@ method_sig_from_accessor_sig (MonoMethodBuilder *mb, gboolean hasthis, MonoMetho
 	ret->hasthis = hasthis;
 	for (int i = 1; i < ret->param_count; i++)
 		ret->params [i - 1] = ret->params [i];
-	memset (&ret->params[ret->param_count - 1], 0, sizeof (MonoType)); // just in case
```

Comment:
This would remove the data for `ret->ret->data.klass` for some reason, which I didn't understand.

Reply:
That's strange. I would expect it to overwrite ret->ret only if the param count is 0. But there's an assert that it's greater than zero. Strange

Reply:
Oh, I see.  `ret->params[0]` is a `MonoType*` not a `MonoType`.

so it should be:

```c
memset (&ret->params[ret->param_count - 1], 0, sizeof (MonoType*)); // just in case
```

otherwise we overwrite too much stuff.  (And in particular `mono_metadata_signature_dup_internal` allocates `ret->ret` right after the last param)